app/Models/LivestreamCollection.php

```diff
@@ -36,17 +36,50 @@ public static function promote($id)
     public function all()
     {
         if ($this->streams === null) {
-            $this->streams = Cache::remember('livestreams:arr', 5, function () {
-                return $this->download()['streams'] ?? [];
+            $this->streams = Cache::remember('livestreams:arr:v2', 5, function () {
+                $streams = $this->downloadStreams()['data'] ?? [];
+
+                $userIds = array_map(function ($stream) {
+                    return $stream['user_id'];
+                }, $streams);
+
+                $users = [];
+
+                foreach ($this->downloadUsers($userIds)['data'] ?? [] as $user) {
+                    $users[$user['id']] = $user;
+                }
+
+                return array_map(function ($stream) use ($users) {
+                    return new Twitch\Stream($stream, $users[$stream['user_id']]);
+                }, $streams);
             });
         }
 
         return $this->streams;
     }
 
-    public function download()
+    public function downloadStreams()
+    {
+        return $this->download('streams?first=40&game_id=21465');
+    }
+
+    public function downloadUsers($userIds)
+    {
+        if (count($userIds) === 0) {
+        }
```

Comment:
Is there supposed to be something inside this if block? 🤔

Reply:
That's a good question. And I can't remember what's this for ?_?